marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -545,7 +551,7 @@ public void afterExecute(Runnable r, Throwable t) {
                 //Run the next step
                 else {
                     if(!(threadPool != null && threadPool.isTerminating())) {
```

Comment:
It does not relate to PR. But the condition does not seem correct. 

```suggestion
                    if (!threadPool != null && !threadPool.isTerminating()) {
```

Reply:
I think that's correct, because it avoids a null-pointer issue if the threadPool is null, since the next line tries to do something with the threadPool.

Reply:
@rjrudin if threadpool is null, then the current condition is definitly true. It will produce NullPointerExeption! It is a old bug! If the threadpool is not null and it is not terminating, we want to continue execute next task.